ui/pages/hyperliquid-referral-consent/hyperliquid-referral-consent.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState } from 'react';
+import {
+  Box,
+  Button,
+  Checkbox,
+  Text,
+} from '../../components/component-library';
+import {
+  AlignItems,
+  BackgroundColor,
+  BlockSize,
+  BorderRadius,
+  Display,
+  FlexDirection,
+  FontWeight,
+  JustifyContent,
+  TextColor,
+  TextVariant,
+} from '../../helpers/constants/design-system';
+import { useI18nContext } from '../../hooks/useI18nContext';
+
+const HYPERLIQUID_REFERRAL_LEARN_MORE_URL =
+  'https://hyperliquid.gitbook.io/hyperliquid-docs/referrals';
+
+export type HyperliquidReferralConsentProps = {
+  onActionComplete: (result: {
+    approved: boolean;
+    allAccounts: boolean;
+    selectedAddress: string;
+  }) => void;
+  allAccounts: boolean;
+  selectedAddress: string;
+};
+
+export const HyperliquidReferralConsent: React.FC<
+  HyperliquidReferralConsentProps
+> = ({ onActionComplete, allAccounts, selectedAddress }) => {
+  const t = useI18nContext();
+  const [isChecked, setIsChecked] = useState(true);
+
+  const handleSubmit = () => {
+    onActionComplete({
+      approved: isChecked,
+      allAccounts,
+      selectedAddress,
+    });
+  };
+
+  const handleCheckboxClick = () => {
+    setIsChecked(!isChecked);
+  };
+
+  return (
+    <Box
+      display={Display.Flex}
+      flexDirection={FlexDirection.Column}
+      height={BlockSize.Full}
+    >
+      <Box
```

Comment:
Minor, as you have a lot of nested boxes, is it worth creating some local sub-components such as `Hyperlink`, `Image` to improve readability?